### PR TITLE
BREAK: remove 2.7 support; remove docker; use travis python

### PR DESCRIPTION
**Pull request by aikikode**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,7 +1,5 @@
 # This Makefile requires the following commands to be available:
 # * python3.6
-# * docker
-# * docker-compose
 
 DEPS:=requirements.txt
 DOCKER_COMPOSE:=$(shell which docker-compose)
```
Comment (kammala) [inline Makefile]: also looks obsolete

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,14 @@
...
+python:
+  - "3.5"
+  - "3.6"
+  - "3.7"
+  - "3.8"
+script: tox
 before_install:
 - pip install codecov
+install:
+- pip install tox-travis
```
Comment (kammala) [inline .travis.yml]: should we merge it with codecov?

**Review by kammala — DISMISSED** [review]

File: setup.py [hunk 3]
```diff
@@ -26,8 +26,6 @@
         "Intended Audience :: Developers",
         "Operating System :: OS Independent",
         "Programming Language :: Python",
-        "Programming Language :: Python :: 2",
-        "Programming Language :: Python :: 2.7",
         "Programming Language :: Python :: 3",
         "Programming Language :: Python :: 3.5",
```
Comment (kammala) [inline setup.py]: shall we drop this one as well?

File: requirements.txt [hunk 4]
```diff
@@ -11,9 +12,10 @@ wheel
 mypy; python_version >= '3.5'
```
Comment (kammala) [inline requirements.txt]: python version specifier might be dropped as well

**Review by kammala — DISMISSED** [review]

**Review by kammala — APPROVED** [review]

**Review by ricardosantosalves — APPROVED** [review]

**Pull request merged**